### PR TITLE
Sparkdoc

### PR DESCRIPTION
**Pull request by zjf2012**
added documentation for how to use Hadoop DAOS in spark. 

This PR also includes some correction to perf test as well as more debug logs.

### CODE REVIEW

File: doc/user/spark.md [hunk 1]
```diff
@@ -1,4 +1,115 @@
...
+    $ cd daos
+    $ git checkout <desired branch or commit>
+    ## assume DAOS is built and installed to <daos_install> directory
+    $ cd src/client/java
+    $ mvn clean package -DskipITs -Ddaos.install.path=<daos_install>
+
+After build, the package daos-java-<version>-assemble.tgz will be available under distribution/target.
+
+## Deploy DAOS Hadoop Filesystem Jars
+After unzipping daos-java-<version>-assemble.tgz, you’ll get the following files.
```
Comment (johannlombardi) [inline doc/user/spark.md]: You need to add a blank line before bullets otherwise mkdocs does not recognize it.

File: doc/user/spark.md [hunk 2]
```diff
@@ -1,4 +1,115 @@
...
+* daos-java-api-<version>.jar and hadoop-daos-<version>.jar
+These files need to be deployed on every compute node that runs Spark. Place them in a directory, e.g.,
+$SPARK_HOME/jars, that are accessible to all the nodes or copy them to every node.
+
+* daos-site-example.xml
+The file contains DAOS configuration and needs to be properly configured with the DAOS pool UUID, container UUID and a
+few other settings. Rename it to daos-site.xml and place it in Spark’s conf ($SPARK_HOME/conf) directory.
+
+## Configure Spark to use DAOS
```
Comment (johannlombardi) [inline doc/user/spark.md]: Same here.

File: src/client/SConscript [hunk 3]
```diff
@@ -9,6 +9,7 @@ def scons():
     SConscript('dfs/SConscript')
     SConscript('dfuse/SConscript')
     SConscript('pydaos/SConscript')
+    SConscript('java/SConscript')
```
Comment (johannlombardi) [inline src/client/SConscript]: hm, i don't see where java/SConscript is being added. Did you forgot to git add?

File: .travis.yml [hunk 4]
```diff
@@ -7,10 +7,13 @@ env:
 
 language: c
 
+language: java
+jdk: openjdk8
```
[the remark this answers is not in the document]
Comment (zjf2012) [reply .travis.yml]: not yet. it's preparation for java CI build.

File: doc/user/spark.md [hunk 5]
```diff
@@ -1,4 +1,115 @@
-## Spark Support
+#Get Started with DAOS Hadoop Filesystem
```
[the remark this answers is not in the document]
Comment (zjf2012) [reply doc/user/spark.md]: done

File: src/client/SConscript (hunk 3 above)
Comment (zjf2012) [reply src/client/SConscript]: it's just a preparation. I'll complete it in next PR.

File: doc/user/spark.md (hunk 1 above)
Comment (zjf2012) [reply doc/user/spark.md]: I just checked the rendering of spark.md in the "sparkdoc" branch in github. it works as expected. Is there any misunderstanding?

**Comment by johannlombardi** [comment]
@cdurf1 please review the doc section.

File: src/client/SConscript (hunk 3 above)
Comment (johannlombardi) [reply src/client/SConscript]: ok, and it does not break the build meanwhile to invoke a SConscript that does not exist?

Comment (zjf2012) [reply src/client/SConscript]: sorry. it's failed. I overlooked the email. let me roll it back. thanks

File: .travis.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (zjf2012) [reply .travis.yml]: let me remove it too.

**Comment by zjf2012** [comment]
removed incomplete java CI build.

File: doc/user/spark.md (hunk 1 above)
Comment (johannlombardi) [reply doc/user/spark.md]: we use mkdocs and i am afraid that it has a different interpreter. Anyway, i can fix that when updating the online site.

**Comment by cdurf1** [comment]
> @cdurf1 please review the doc section.

I will need ~ 30 more minutes for review

**Pull request merged**